frame/staking/src/lib.rs

```diff
@@ -3127,6 +3139,42 @@ impl<T: Trait> Module<T> {
 		})
 	}
 
+	/// Use a damping function to update the target number of validators for the next selection.
+	fn update_target_validators(mut exposures: Vec<BalanceOf<T>>) -> u32 {
+		let average = |all: &[BalanceOf<T>]| -> BalanceOf<T> {
+			all.iter().fold(Zero::zero(), |total: BalanceOf<T>, x: &BalanceOf<T>| {
+				// `total` should never saturate since total issuance fits inside `Balance`
+				total.saturating_add(*x)
+			}) / BalanceOf::<T>::saturated_from(all.len() as u128)
```

Comment:
I think we assume `x.len() as u32` in runtime and then you can just `into()` it to `BalanceOf<T>`.

Reply:
i.e. 

```
```suggestion
			}) / BalanceOf::<T>::from(all.len() as u32)
```

Reply:
would be good to be safe against len == 0 as well.